packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -345,6 +346,61 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
             }
         }
 
+        private static IReadOnlyList<ParameterProvider> BuildOrderedSpreadParameters(InputModelType inputModel)
+        {
+            // create the model provider
+            if (ClientModelPlugin.Instance.TypeFactory.CreateModel(inputModel) is not ModelProvider modelProvider)
+                throw new InvalidOperationException($"Unable to create model for input model {inputModel.Name}.");
```

Comment:
So I think this suggests that if some one is trying to overhaul the model provider with their own, they will have to override the methods/clients providers as well?

Reply:
@ArcturusZhang that's correct. The motivation behind that was to use the ModelProvider's FullConstructor property that already does all the work of building the constructor signature using all the properties of the model. We _could_ remove this check on modelprovider and keep the type flexible, and either use the model's properties directly or try to determine which constructor is the "full constructor" to build the new constructor for the spread. That would avoid this additional place a user would have to override.

Reply:
@chunyu3 - type factory's `CreateModel` is idempotent so it returns the cached modelprovider if it was already created prior.